integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -1256,58 +1256,40 @@ fn test_bad_orphan() {
 
 #[test]
 fn test_bad_chunk_mask() {
-    // TODO(#10506): Fix test to handle stateless validation
-    if checked_feature!("stable", StatelessValidationV0, PROTOCOL_VERSION) {
-        return;
-    }
-
     init_test_logger();
-    let chain_genesis = ChainGenesis::test();
-    let validators = vec!["test0".parse().unwrap(), "test1".parse().unwrap()];
-    let mut clients: Vec<Client> = validators
-        .iter()
-        .map(|account_id| {
-            let vs = ValidatorSchedule::new()
-                .num_shards(2)
-                .block_producers_per_epoch(vec![validators.clone()]);
-            setup_client_with_synchronous_shards_manager(
```

Comment:
In a follow up PR, do you think it's possible to completely remove all reference of `setup_client_with_synchronous_shards_manager`? This function uses the dreaded mock epoch manager and is only used in the set of tests in this file.

Reply:
Never mind, I'll follow up on this

Reply:
Here we are! https://github.com/near/nearcore/pull/10586